File: Java/board/service/comment/src/main/java/board/comment/controller/CommentController.java
package board.comment.controller;

import board.comment.service.CommentService;
import board.comment.service.request.CommentCreateRequest;
import board.comment.service.response.CommentResponse;
import board.comment.service.response.PageResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequiredArgsConstructor
@RequestMapping("/v1/comments")
public class CommentController {
    private final CommentService commentService;

    @GetMapping("/{commentId}")
    public CommentResponse read(
            @PathVariable("commentId") Long commentId
    ) {
        return commentService.read(commentId);
    }

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public CommentResponse create(@RequestBody CommentCreateRequest request) {
        return commentService.create(request);
    }

    @DeleteMapping("/{commentId}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public void delete(@PathVariable("commentId") Long commentId) {
        commentService.delete(commentId);
    }

    @GetMapping
    public PageResponse<CommentResponse> readAll(
            @RequestParam("articleId") Long articleId,
            @RequestParam("page") Long page,
            @RequestParam("pageSize") Long pageSize
    ) {
        return commentService.readAll(articleId, page, pageSize);
    }

    @GetMapping("/infinite-scroll")
    public List<CommentResponse> readAll(
            @RequestParam("articleId") Long articleId,
            @RequestParam(value = "lastParentCommentId", required = false) Long lastParentCommentId,
            @RequestParam(value = "lastCommentId", required = false) Long lastCommentId,
            @RequestParam("pageSize") Long pageSize
    ) {
        return commentService.readAll(articleId, lastParentCommentId, lastCommentId, pageSize);
    }
}

File: Project/blog/backend/src/main/kotlin/com/blog/backend/controller/CommentController.kt
package com.blog.backend.controller

import com.blog.backend.controller.dto.CommentDto
import com.blog.backend.controller.dto.CreateCommentDto
import com.blog.backend.controller.dto.PageDto
import com.blog.backend.service.CommentService
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/v1")
class CommentController(
    private val commentService: CommentService,
) {

    @GetMapping("/comments/by-post-id/{postId}/page/{page}")
    fun findPagedCommentByPostId(
        @PathVariable postId: Long,
        @PathVariable page: Int,
        @RequestParam size: Int,
    ): PageDto<CommentDto> {
        return commentService.findPagedCommentByPostId(page, size, postId)
    }

    @PostMapping("/comments")
    fun createComment(
        @RequestBody dto: CreateCommentDto,
    ): CommentDto {
        return commentService.createComment(dto)
    }

    @ResponseStatus(HttpStatus.NO_CONTENT)
    @DeleteMapping("/comments/{id}")
    fun deleteById(
        @PathVariable id: Long,
    ) {
        commentService.deleteCommentById(id)
    }

}
